src/controllers/payment.controller.ts

```diff
@@ -24,8 +29,9 @@ export class PaymentController extends BaseController {
   public async get(): Promise<RedirectResult> {
     const token: string = this.session.getAccessToken(this.httpContext.request)
     const dissolutionSession: DissolutionSession = this.session.getDissolutionSession(this.httpContext.request)!
+    const dissolution: Optional<DissolutionGetResponse> = await this.getDissolution(dissolutionSession, token)
 
-    if (dissolutionSession.isApplicationAlreadyPaid) {
+    if (dissolution!.application_status === ApplicationStatus.PAID) {
```

Comment:
i think you can change this to:
```
if (isAlreadyPaid(dissolutionSession, token)) {
  return this.redirect(SEARCH_COMPANY_URI)
}
```

then you can move the dissolution retrieval out of the public method to keep it light and readable